contracts/RewardPool.sol

```diff
@@ -92,13 +92,39 @@ contract RewardPool is Initializable, Versionable, Ownable {
     return true;
   }
 
+  function valid(
+    bytes calldata leaf,
+    bytes calldata proof
```

Comment:
im assuming here that proof does not have meta appended to it?

Reply:
I notice you combined everything into leaf. So even things that were not in the original leaf such as `paymentCycle` was just included into leaf.

Reply:
Yeah, my idea was to have two things, the leaf node and the proofs. Leaf is now kind of what was the meta & the calling arguments, and in my mind should contain everything needed to verify it's valid. 

Requires a change elsewhere obviously.